realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -98,6 +100,21 @@ public void onError(SyncSession session, ObjectServerError error) {
     private static volatile AuthenticationServer authServer = new OkHttpAuthenticationServer();
     private static volatile UserStore userStore;
 
+    private static NetworkStateReceiver.ConnectionListener networkListener = new NetworkStateReceiver.ConnectionListener() {
+        @Override
+        public void onChange(boolean connectionAvailable) {
+            if (connectionAvailable) {
+                if (NetworkStateReceiver.isOnline(SyncObjectServerFacade.getApplicationContext())) {
```

Comment:
This checking seems not necessary? See https://github.com/realm/realm-java/blob/79152436a06c7fd92806483fa4f07eb9a675702b/realm/realm-library/src/objectServer/java/io/realm/internal/network/NetworkStateReceiver.java#L82